core/src/com/mygdx/minigolf/controller/systems/GraphicsSystem.java

```diff
@@ -0,0 +1,102 @@
+
+    // Number of pixels per meter
+    private static final float PPM = 32.0f;
+
+    // Height and width of camera frustum, based off width and height of the screen and pixel per meter ratio
+    private static final float FRUSTUM_WIDTH = Gdx.graphics.getWidth() / PPM;
+    private static final float FRUSTUM_HEIGHT = Gdx.graphics.getHeight() / PPM;
+
+    private final OrthographicCamera cam = new OrthographicCamera(FRUSTUM_WIDTH, FRUSTUM_HEIGHT);
+
+    private final Array<Entity> renderQueue = new Array<>(); // A sorted array of entities based on level
+    private final Comparator<Entity> comparator = new LayerComparator(); // A comparator to sort entities based on their level
+
+    private final ComponentMapper<Physical> physicalMapper = ComponentMapper.getFor(Physical.class);
+    private final ComponentMapper<Graphical> graphicalMapper = ComponentMapper.getFor(Graphical.class);
+
+    // A shape renderer used for testing purpose (not using textures)
+    private final ShapeRenderer shapeRenderer = new ShapeRenderer();
+
+    public GraphicsSystem() {
+        super(Family.all(Physical.class, Graphical.class).get(), new LayerComparator());
+
+        cam.position.set(FRUSTUM_WIDTH / 2f, FRUSTUM_HEIGHT / 2f, 0);
+    }
+
+    @Override
+    public void update(float dt) {
+        super.update(dt);
+
+        // Sort the renderQueue based on layer
+        renderQueue.sort(comparator);
+
+        cam.update();
+
+        // Render shapes
+        shapeRenderer.setProjectionMatrix(cam.combined);
+        shapeRenderer.setAutoShapeType(true);
+        shapeRenderer.begin();
+        shapeRenderer.setColor(Color.WHITE);
+
+        for (Entity entity : renderQueue) {
+            Physical physical = physicalMapper.get(entity);
+
+            switch (physical.getShape().getType()) {
+                case Circle:
+                    shapeRenderer.set(ShapeRenderer.ShapeType.Filled);
+                    shapeRenderer.circle(physical.getPosition().x, physical.getPosition().y, physical.getShape().getRadius(), 50);
+                    break;
+                case Polygon:
+                    shapeRenderer.set(ShapeRenderer.ShapeType.Line);
+                    PolygonShape polygonShape = (PolygonShape) physical.getShape();
+
+                    Vector2 vertex1 = new Vector2();
+                    Vector2 vertex2 = new Vector2();
+
+                    for (int i = 0; i < polygonShape.getVertexCount(); i++) {
+                        polygonShape.getVertex(i, vertex1);
+                        if (i == 0) {
+                            polygonShape.getVertex(polygonShape.getVertexCount() - 1, vertex2);
+                        } else {
+                            polygonShape.getVertex(i - 1, vertex2);
+                        }
+                        vertex1.add(physical.getPosition());
+                        vertex2.add(physical.getPosition());
+                        shapeRenderer.line(vertex2, vertex1);
+                    }
```

Comment:
Muligens en ikke-optimal måte å rendre polygons på (rendrer linje for linje og de blir ikke fylt med farge). Prøvde først å bruke `shapeRenderer.polygon(float[] vertices)`, men som man ser tar den inn en array literal av den primitive typen float. Slet med å konvertere listen med Vector2-objekter til en slik array literal.  Listen med Vector2-objekter fikk jeg ved å loope gjennom `polygonShape.getVertex(i, vertex)` for hver vertex index (ja, det er ingen metode for å hente alle 😢). Jeg så også på `PolygonSprite(PolygonRegion region) `, men for å lage PolygonRegion trenger man også en float[] som for shapeRenderer.

Tar gjerne imot tanker og evt. forslag til hvordan det kan gjøres!